### ISSUE TITLE
Continual Backprop: Stochastic Gradient Descent with Persistent Randomness

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#303]
### 論文リンク
https://arxiv.org/abs/2108.06325

## 概要
ニューラルネットワークの学習アルゴリズムであるBackpropは、確率的勾配降下と小さなランダムな重みを用いた初期化という2つのメカニズムを用いており、後者は前者の有効性を高めるために不可欠である。我々は、継続的な学習において、Backpropは初期には良好な性能を示すが、時間の経過とともに性能が低下することを示す。確率的勾配降下法のみでは継続的な学習には不十分であり、初期のランダム性は初期学習のみを可能にし、継続的な学習は不可能である。このようなBackpropの学習能力の劣化を示す結果は、我々の知る限り、今回が初めてです。このようなBackpropの可塑性の低下を解決するために、我々は新しい生成-テストプロセスを用いて、勾配降下と同時にランダムな特徴を継続的に注入するアルゴリズムを提案します。このアルゴリズムを "textit{Continual Backprop}"アルゴリズムと呼ぶ。Backpropとは異なり、Continual Backpropは教師あり学習と強化学習（RL）の両方の問題で継続的に適応できることを示す。Continual BackpropはBackpropと同じ計算量であり、Backpropの継続的学習に対する自然な拡張と見なすことができる。

## 実験方法・結果

<img width="342" alt="Screenshot 2022-10-12 at 6 58 52 PM" src="https://user-images.githubusercontent.com/8721858/195463127-c18df804-6827-473b-a9aa-37cc67b9d5dc.png">
<img width="335" alt="Screenshot 2022-10-12 at 6 59 00 PM" src="https://user-images.githubusercontent.com/8721858/195463129-710f4437-38ef-4f43-890a-4b42c929a3a6.png">
<img width="332" alt="Screenshot 2022-10-12 at 6 59 09 PM" src="https://user-images.githubusercontent.com/8721858/195463131-2e04fd33-ab16-406c-ae3d-3b9abbe479d4.png">
<img width="666" alt="Screenshot 2022-10-12 at 6 59 16 PM" src="https://user-images.githubusercontent.com/8721858/195463132-e8e12c8a-d248-4398-8399-98990c470406.png">
<img width="344" alt="Screenshot 2022-10-12 at 6 59 24 PM" src="https://user-images.githubusercontent.com/8721858/195463135-16f025cf-3fa7-45ae-8f30-cfea1d910f08.png">
